libs/components/src/alert/vl-alert.component.ts

```diff
@@ -1,167 +1,94 @@
+import { BaseLitElement } from '@domg-wc/common-utilities';
 import { alertStyle, iconStyle } from '@domg/govflanders-style/component';
-import { accessibilityStyle, resetStyle } from '@domg/govflanders-style/common';
-import { VlAlertClosedEvent } from './vl-alert-closed-event';
-
-@webComponent('vl-alert')
-export class VlAlert extends BaseElementOfType(HTMLElement) {
-    static get _observedAttributes() {
-        return ['icon', 'title', 'closable', 'type', 'size'];
-    }
-
-    constructor() {
-        super(`
-          <style>
-            ${resetStyle}
-            ${iconStyle}
-            ${alertStyle}
-            ${accessibilityStyle}
-          </style>
-          <div id="alert" class="vl-alert" role="alert">
-            <div id="content" class="vl-alert__content">
-              <p id="title" class="vl-alert__title">
-                <slot name='title'></slot>
-              </p>
-              <div id="message" class="vl-alert__message">
-                <slot id="messages-slot"></slot>
-              </div>
-              <div id="actions" class="vl-alert__actions">
-                <slot id="actions-slot" name="actions"></slot>
-              </div>
+import { accessibilityStyle, resetStyle, markStyle } from '@domg/govflanders-style/common';
+import { VlAlertClosedEvent } from './vl-alert.model';
+import { classMap } from 'lit/directives/class-map.js';
+
+@customElement('vl-alert')
+export class VlAlert extends BaseLitElement {
+    icon = '';
+    title = '';
+    type = '';
+    size = '';
+    message = '';
+    naked = false;
+    closable = false;
+
+    static get styles(): CSSResult[] {
+        return [resetStyle, alertStyle, iconStyle, accessibilityStyle, markStyle];
+    }
+
+    static get properties(): PropertyDeclarations {
+        return {
+            icon: { type: String, attribute: 'data-vl-icon', reflect: true },
+            title: { type: String, attribute: 'data-vl-title', reflect: true },
+            closable: { type: Boolean, attribute: 'data-vl-closable', reflect: true },
+            type: { type: String, attribute: 'data-vl-type', reflect: true },
+            size: { type: String, attribute: 'data-vl-size', reflect: true },
+            naked: { type: Boolean, attribute: 'data-vl-naked', reflect: true },
+            message: { type: String, attribute: 'data-vl-message', reflect: true },
+        };
+    }
+
+    protected updated(changedProperties: Map<string, unknown>): void {
+        super.updated(changedProperties);
+        this.processButtons();
+    }
+
+    protected render(): TemplateResult {
+        const classes = {
+            'vl-alert': true,
+            [`vl-alert--${this.type}`]: Boolean(this.type),
+            'vl-alert--small': this.size === 'small',
+            'vl-alert--naked': this.naked,
+        };
+
+        const markClass = this.naked ? `vl-u-mark--${this.type}` : '';
+
+        return html`
+            <div id="alert" class=${classMap(classes)} role="alert">
```

Comment:
op zich kan je hier klasses die er altijd op staan gewoon al meegeven;

dus ipv die op `true` te zetten, kan je ook gewoon dit doen:

`class="vl-alert ${classMap(classes)}"`

zie Lit docs: https://lit.dev/docs/templates/directives/#classmap

Reply:
Goed om te weten. Maar persoonlijk ben ik meer voor het definiëren van de klassen op 1 plaats ipv op 2 dus ik ga het in dit geval zo laten.